flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendRunTest.java

```diff
@@ -130,6 +131,46 @@ public void testRun() throws Exception {
         }
     }
 
+    @Test
+    public void testClaimRestoreModeParsing() throws Exception {
+        // test configure savepoint with claim mode
+        String[] parameters = {
+            "-s", "expectedSavepointPath", "-n", "-r", "claim", getTestJarPath()
```

Comment:
```
-s expectedSavepointPath -n -r claim
-s expectedSavepointPath -n -r no-claim
```
? Is this our convention? `-r` It's really cryptic and hard to remember/understand. Isn't this better:
```
-s expectedSavepointPath -n -claim
-s expectedSavepointPath -n -no-claim
```
or
```
-s expectedSavepointPath -n -restore-mode claim
-s expectedSavepointPath -n -restore-mode no-claim
```
?

Reply:
```
-s expectedSavepointPath -n -restore-mode claim
-s expectedSavepointPath -n -restore-mode no-claim
```
works as well (`-r` is just a shortcut for `-restore-mode`)

The argument against `-claim` or `-no-claim` was that it does not work well for other modes such as e.g. `legacy`.

Reply:
If you think that's confusing I guess we can drop `-r` and stick with just `restoreMode`. It would also solve @Myasuka concern about duplicated `-r` option.

Reply:
> The argument against -claim or -no-claim was that it does not work well for other modes such as e.g. legacy.

Could you elaborate what do you mean? Why it does not work?

Reply:
Using `-s expectedSavepointPath -n --legacy` is not necessarily descriptive. It is not straightforward to tell what the `legacy` refers to.

This has been also pointed out by @rkhachatryan. Do you have any comments on that?

Reply:
Ok, I see. I would personally be in favour of dropping the `-r` abbreviation. IMO Those commands are used so rarely, that it doesn't make sense to abbreviate them sacrificing readability. They will be either used once a year, or written once in a script/code and then read many times. In both cases abbreviation would do more harm than good.

Reply:
I agree, keeping only `-restoreMode` without abbrevitions and with several values should suffice and be less confusing IMO.

Reply:
In the current state of the PR, `-r` is no longer valid.